opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -388,19 +389,111 @@ namespace Opm {
+                    break;
                 }
             }
+
+
+            const std::string& wellNamePattern = record->getItem("WELL")->getTrimmedString(0);
+            const std::vector<WellPtr>& wells = getWells(wellNamePattern);
+
             WellCommon::StatusEnum status = WellCommon::StatusFromString( record->getItem("STATUS")->getTrimmedString(0));
-            well->setStatus(currentStep, status);
+
+            for (auto wellIter=wells.begin(); wellIter != wells.end(); ++wellIter) {
+                WellPtr well = *wellIter;
+                std::string wellName = well->name();
+                if(haveCompletionData){
+                    CompletionSetConstPtr currentCompletionSet = well->getCompletions(currentStep);
+
+                    CompletionSetPtr newCompletionSet(new CompletionSet( ));
+
+                    Opm::Value<int> I  = getValueItem(record->getItem("I"));
+                    Opm::Value<int> J  = getValueItem(record->getItem("J"));
+                    Opm::Value<int> K  = getValueItem(record->getItem("K"));
+                    Opm::Value<int> C1 = getValueItem(record->getItem("C1"));
+                    Opm::Value<int> C2 = getValueItem(record->getItem("C2"));
+
+                    size_t completionSize = currentCompletionSet->size();
+
+                    for(size_t i = 0; i < completionSize;i++) {
+
+                        CompletionConstPtr completion = currentCompletionSet->get(i);
+
+                        if (C1.hasValue()) {
+                            if (i < (size_t) C1.getValue()) {
+                                newCompletionSet->add(completion);
+                                continue;
+                            }
+                        }
+                        if (C2.hasValue()) {
+                            if (i > (size_t) C2.getValue()) {
+                                newCompletionSet->add(completion);
+                                continue;
+                            }
+                        }
+
+                        int ci = completion->getI();
+                        int cj = completion->getJ();
+                        int ck = completion->getK();
+
+                        if ((I.hasValue() && (!(I.getValue() == ci) || I.getValue() !=0))) {
+                            newCompletionSet->add(completion);
+                            continue;
+                        }
+
+                        if ((J.hasValue() && (!(J.getValue() == cj) || J.getValue()!=0))) {
+                            newCompletionSet->add(completion);
+                            continue;
+                        }
+
+                        if ((K.hasValue() && (!(K.getValue() == ck) || K.getValue()!=0))) {
+                            newCompletionSet->add(completion);
+                            continue;
+                        }
+
+                        Value<double> transmissibilityFactor("transmissibilityFactor", completion->getConnectionTransmissibilityFactor());
```

Comment:
Instead of the code block on lines 473 - 488 I would suggest adding a copy constructor: 
```C++
Completion::Completion(const Completion& oldCompletion, WellCompletion::StatusEnum newStatus) {
    ...
}
```

Also observe that when creating new shared pointers the preferred way is to use `std::make_shared<T>( ... )`